packages/ra-core/src/controller/field/ReferenceManyField.stories.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import {
+    CoreAdminContext,
+    type CoreAdminContextProps,
+    type GetManyResult,
+    type ListControllerResult,
+    testDataProvider,
+    useReferenceManyFieldController,
+} from '../..';
+
+const defaultDataProvider = testDataProvider({
+    getManyReference: (_resource, params): Promise<GetManyResult> =>
+        Promise.resolve({
+            data: [
+                { id: 0, title: 'bar0' },
+                { id: 1, title: 'bar1' },
+            ].slice(0, params.pagination.perPage),
+            total: params.pagination.perPage || 2,
+        }),
+});
+
+const ReferenceManyFieldController = props => {
+    const { children, ...rest } = props;
+    const controllerProps = useReferenceManyFieldController({
+        sort: {
+            field: 'id',
+            order: 'ASC',
+        },
+        ...rest,
+    });
+    return children(controllerProps);
+};
+
+const ReferenceManyFieldComponent = (props: ListControllerResult) => (
+    <div>
+        <div
+            style={{
+                display: 'flex',
+                alignItems: 'center',
+                gap: '10px',
+            }}
+        >
+            <button
+                onClick={() => props.onSelectAll()}
+                disabled={props.total === props.selectedIds.length}
+            >
+                Select All
+            </button>
+            <button
+                onClick={props.onUnselectItems}
+                disabled={props.selectedIds.length === 0}
+            >
+                Unselect All
+            </button>
+            <p>Selected ids: {JSON.stringify(props.selectedIds)}</p>
+        </div>
+        <ul
+            style={{
+                listStyleType: 'none',
+            }}
+        >
+            {props.data?.map(record => (
+                <li key={record.id}>
+                    <input
+                        type="checkbox"
+                        checked={props.selectedIds.includes(record.id)}
+                        onChange={() => props.onToggleItem(record.id)}
+                        style={{
+                            cursor: 'pointer',
+                            marginRight: '10px',
+                        }}
+                    />
+                    {record.id} - {record.title}
+                </li>
+            ))}
+        </ul>
+    </div>
+);
+
+export const ReferenceManyField = ({
```

Comment:
I'd call it `Basic` instead

Reply:
I didn't agree. By naming it `Basic`, our Storybook's tree structure will be this one:

![image](https://github.com/user-attachments/assets/ee02c3c6-0af0-47c2-a1df-15777447cf1d)

instead of this one:

![image](https://github.com/user-attachments/assets/df266408-9b71-4be2-b449-29a5d7fcc9be)

I'm not sure that a drop down for only one item is useful

Reply:
I'd do it anyway as this component obviously don't have enough stories and we should/will add more (not in this PR though)